File: DemoServer/AdditionalLanguages/python/relatedDocuments/queryRelatedDocuments/queryRelatedDocuments.py
from demo_example import Example, RunParamsBase
from models import Order, Product

class QueryRelatedDocuments(Example):
    def run(self, run_params: RunParamsBase = None) -> None:
        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_1
            shipped_orders = list(session.query(object_type = Order)\
                                  .include("Lines.Product")\
                                  .where_exists("ShippedAt"))
            #endregion

            #region Step_2
            for shipped_order in shipped_orders:
                product_ids = [x.Product for x in shipped_order.Lines]
                #endregion

                for i in range(len(product_ids)):
                    #region Step_3
                    product = session.load(product_ids[i], Product)
                    product.UnitsOnOrder = product.UnitsOnOrder + shipped_order.Lines[i].Quantity
                    #endregion

            #region Step_4
            session.save_changes()
            #endregion
        #endregion

File: DemoServer/AdditionalLanguages/php/relatedDocuments/queryRelatedDocuments/QueryRelatedDocuments.php
<?php

namespace RavenDB\Demo\relatedDocuments\queryRelatedDocuments;

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Order;
use RavenDB\Demo\common\models\Product;

class QueryRelatedDocuments
{
    public function __invoke(): string
    {
        //region Demo
        $session = DocumentStoreHolder::getStore()->openSession();
        try {
            //region Step_1
            $shippedOrders = $session->query(Order::class)
                ->include("Lines.Product")
                ->whereExists("ShippedAt")
                ->toList();
            //endregion

            //region Step_2
            /** @var Order $shippedOrder */
            foreach ($shippedOrders as $shippedOrder) {
                $productIds = array_map(
                    function ($x) {
                        return $x->getProduct();
                    },
                    $shippedOrder->getLines()->getArrayCopy()
                );
            //endregion

                for ($i = 0; $i < count($productIds); $i++) {
                    //region Step_3
                    $product = $session->load(Product::class, $productIds[$i]);
                    $product->setUnitsOnOrder($product->getUnitsOnOrder() + $shippedOrder->getLines()[$i]->getQuantity());
                    //endregion
                }
            }

            //region Step_4
            $session->saveChanges();
            //endregion
        } finally {
            $session->close();
        }
        //endregion

        return "The Product Documents were updated successfully";
    }
}
